### ISSUE TITLE
CCBY license for the problem maps (free content)

### ISSUE THREAD

**Issue by prototyperspective** [amelioro/ameliorate#558]
It would be great if you could set the license of the problem trees to [CCBY4.0](https://creativecommons.org/licenses/by/4.0/).

I think this is as important as whether or not the software is open source. Wikipedia and StackExchange are so useful and partly so successful because their contents can be freely reused (also under CCBY licenses).

Having the problem trees be CCBY would enable:
* Uploading image export versions of problem maps to Wikimedia Commons, the global free media repository (companion project to Wikipedia). (I could only upload a screenshot of a map fully created by me to Commons not of e.g. [this](https://ameliorate.app/examples/ontology) or [this](https://ameliorate.app/examples/climate-change) because I can license my content under that license and the software is open source.) It would thereby also enable embedding the problem maps in relevant Wikipedia articles in some cases and other Wiki sites (like talk pages, Wikidata, Wikiversity, and so on).
* Use of the problems trees such as by organizations on their website, news articles (maybe at some point), in youtube videos (including by official organizations that use only media they have permission for as well as CCBY-licensed YT videos), other software (if you're concerned about that use it could also be CCBYSA) including additional clients for the same content, infographics, research study papers, and so on.

Here is an infographic by Martin Owens about [free content](https://en.wikipedia.org/wiki/Free_content):
![Foss-creative-commonssm-print_1](https://github.com/user-attachments/assets/54e32f8b-dbf9-45fc-9d69-fb249733bbf8)

If a CCBY license is used (not [CC0](https://creativecommons.org/public-domain/cc0/)) this means people still need to (and are explicitly required!) to credit the source which could be done by linking to the problem map.

Licensing the content under CCBY4.0 would require a statement that the site's content is under CCBY somewhere on the site and the earlier that's done the better it would be. One could add it to the footer like it's done for Wikimedia Commons, Stackoverflow, and Wikipedia or by adding a note like "_The site's contents are licensed under [CCBY 4.0](https://creativecommons.org/licenses/by/4.0/) unless specified otherwise_" to a relevant subsite. One could also specify that this is only for public problem trees, not problem trees that are private (or display something like "This problem map is not under CCBY 4.0 unless it is made public" somewhere at private problem maps) and display a note about this license when the user tries turn a private problem map to public visibility. Again, this is really important.

**Comment by keyserj** [comment]
> set the license of the problem trees to CCBY4.0

> CCBY would enable [...]

> could also specify that this is only for public problem trees, not problem trees that are private

Thanks for the explanation of why this is useful. This generally sounds good to me. I think there could be usages in the future where people prefer a more-restrictive license (maybe people making more opinionated maps), but I think I don't really care to address those cases at this point. Maybe in the future there could be a separate field to control the license, kind of like how YouTube lets you choose per-video.

> If a CCBY license is used (not CC0) this means people still need to (and are explicitly required!) to credit the source which could be done by linking to the problem map.

This seems good to me.

> Licensing the content under CCBY4.0 would require [...]

This is also very good to know.

Here are some images of example licensing UX in footers:

- StackOverflow (link goes to a [further explanation](https://stackoverflow.com/help/licensing))
![image](https://github.com/user-attachments/assets/3ed0f1ed-a092-4327-b3fd-4d9dab39f64e)

- Wikipedia  
![image](https://github.com/user-attachments/assets/41f4dd34-9676-4618-957f-d74f9deac7dc)

Some questions to figure out:

- how would changing from private -> public, public -> private affect the licensing?
  - seems like a history of visibility would need to be tracked in order to accurately license
  - maybe changing public -> private wouldn't be a big deal because if people had access to the public version, it'd be shareable, but the private version wouldn't be accessible anyway for sharing?
- where to display licensing?
  - the app takes up the whole screen space, so there isn't a footer
  - at the very least, it should display in the Topic settings when visibility is set to Public
  - maybe the diagram (and table) could have a small "licensed under CC BY-SA" link in the corner (similar to the react-flow attribution)
![image](https://github.com/user-attachments/assets/d6753ced-4279-460f-84d0-195ca1adeaff)
- does the site need to have a user/privacy policy before adding this? #192

**Comment by prototyperspective** [comment]
Sounds great!
> how would changing from private -> public, public -> private affect the licensing? 

The content added during it being under CCBY would remain CCBY, the content added after making it private / changing the license would not be CCBY (except if the users who added/modified these nodes have specified somewhere that their contributions are under CCBY). The thing is one can't simply change the license other users' contributions are under by changing a problem map from private -> public / changing the problem map license. It would only affect subsequent contributions. Afaik one can retrospectively change the license from a less permissive license to a more permissive one but that's uncommon and I'm not sure about it, afaik it would work by gaining implicit consent by notifying the contributors about the license and giving them enough time to delete all their contributions if they prefer to not have them be available under CCBY. That's also why things would be much easier if this is sorted out early.
>seems like a history of visibility would need to be tracked in order to accurately license

No, either a problem shrub/map is fully CCBY or it isn't; things in between are probably not really useful (well sometimes if there are just very few other contributions one could manually remove all nonCCBY things but it's certainly not something for which supporting structures would be good or needed). I don't think people would turn public problem maps to private ones. If they do I think what would make more sense was allowing them to clone/fork the map to a private map so it can be used as a base for a private collaboration that is not CCBY.

>if people had access to the public version, it'd be shareable, but the private version wouldn't be accessible anyway for sharing

Exactly. However, it would be good to add a note about the map not being CCBY. I don't know if otherwise people invited to the private map could just publish e.g. a screenshot of a company's private problem map under CCBY based on the license footer/page. Note that this doesn't change that having a public problem map or making it public means the content remains CCBY, turning the map private would affect subsequent additions and usually people would create the problem map with private visibility right away. On Kialo, one can only set the argument map to either Private or Public and then can't change it anymore (note that there users have this [copying/linking functionality](https://support.kialo.com/en/hc/linking-a-discussion-in-another-discussion/) so they can also copy in claims from other argument maps).

>maybe the diagram (and table) could have a small "licensed under CC BY-SA" link in the corner

I think this would be best. And adding it to the FAQ. I like CCBY much more than CCBYSA and whenever I can choose the license when I upload something on Wikimedia Commons I use CCBY. It's much more intercompatible. YouTube only has the option for CCBY so if one wants to use it in a CCBY YT video one could still not use it there. I think one could also not use it in free software projects that are licensed under MIT, Apache, etc since SA means you need to use the same license. I think companies and people using these for purposes that could be called commercial could not use it either as they would need to license things under CCBYSA so if they e.g. create a short film depicting a problem map or write a book containing one or print a heavily modified one on an object it would need to be licensed as CCBYSA as well. Just makes everything way more complicated and less compatible. Maybe there is a case for SA but I don't see a good reason for it. [OurWorldInData](https://ourworldindata.org/) also uses CCBY (see the footer). I think CCBY makes everything easier.

At "licensed under CC BY-SA", the CC BY-SA would need to link to the license page. In the FAQ it could be made clearer like "All site contents are licensed under CCBY-SA except for contributions in private maps".

>does the site need to have a user/privacy policy before adding this?

No, it doesn't.

**Comment by keyserj** [comment]
> one can't simply change the license other users' contributions are under by changing a problem map from private -> public / changing the problem map license. It would only affect subsequent contributions.

This seems like a detail that requires some thought to figure out. If a map switches private -> public and has other contributions in it, then the CCBY licensing in the corner would be incorrect. Potentially a user agreement could say something like "your contribution is CCBY regardless of the map's licensing"? Or "your contribution is licensed the same as the Topic's license, so if the Topic changes to CCBY, your contribution does as well". Or there could extra logic that sends an email out to all contributors with a button like "don't allow my contribution to change to CCBY", but it feels like there has to be a better way than that.

> giving them enough time to delete all their contributions if they prefer to not have them be available under CCBY

I think that once a contribution is approved, the contributor shouldn't necessarily be able to delete that contribution outside of an approval process (assuming the map requires changes to be approved).

> it would be good to add a note about the map not being CCBY

Yeah this seems good to me. I wonder if "not licensed by CCBY" is good enough verbiage, like does "no license" offer intended lack of permissions?

> I like CCBY much more than CCBYSA

> SA means you need to use the same license

Good to know, I wasn't aware of the difference. I agree that CCBY seems like it'd be more suitable for this then.

> the CC BY-SA would need to link to the license page. In the FAQ it could be made clearer

Yeah this seems good

**Comment by keyserj** [comment]
To summarize the expected efforts here:
- if topic is public: "licensed by CCBY" in bottom corner of diagram & table; else: "not licensed by CCBY"
- bottom-corner messaging links to a docs page that explains that a Topic and all its contributions are CCBY if the Topic is public, and are not CCBY if private. Also explains that if switching private -> public, Topic and all contributions become CCBY
- user agreement that at least says "your contributions to Topics are licensed to match the Topic's license: CCBY if public, not CCBY if private. And the license will change if Topic changes private -> public."
- clear/visible sentence on Topic settings page that conveys "CCBY if public, not CCBY if private"

**Comment by prototyperspective** [comment]
> Or "your contribution is licensed the same as the Topic's license, so if the Topic changes to CCBY, your contribution does as well".

Yes, I think that would be the solution: specifying somewhere in an expectable page like Docs/Licensing or Docs/Terms of Service that the user grants the map-owner/-admins the right to change the license of their contributions to CCBY. I don't think this will occur often and don't think an email should be sent when this happens.

>I wonder if "not licensed by CCBY" is good enough verbiage, like does "no license" offer intended lack of permissions?

That would be good enough, I'd just add something like "contributions are owned by their poster", like some sites have it. However, having this only in the FAQ/License page and a tooltip would probably be best so it only is a small "not CCBY" (or "not licensed under CCBY") text somewhere in the corner (it's enough functionally and all of these things could be improved later on when it comes to e.g. wording).

**Comment by prototyperspective** [comment]
>So if a Topic is not CCBY, the poster would own their contributions, but they just don't have rights to prevent the contribution from becoming CCBY if the Topic is changed to CCBY?

They still own them after they are CCBY. You ask good questions, I'm not an expert in this. I think it's fine – if that text clearly says that the map admin(s) can change the license then it's not an issue. I think much of this isn't really clear and specified so there isn't much clarity and guidance. For better clarity I'd ask on a relevant subreddit or at https://opensource.stackexchange.com/ where you could also mention the site. I think the use is so that the people need to be credited which if for example some study or news article quotes a node, they should name its author username or if somebody puts a valuable idea in some node(s) then they have the intellectual ownership of it (but it's still CCBY). They could control how they're being credited but I've rarely seen anybody specify that and also for a many-authors map one wouldn't name all map contributors but just link the map which should have some revision page where people can see who contributed to it (maybe also some stats like [these](https://xtools.wmcloud.org/articleinfo/en.wikipedia.org/Science) at some point). If you don't ask about it maybe I will, it's not unlikely there hasn't been any other case like it so far so it would be good to know but it also wouldn't be a problem to just specify this in the License page (maybe link to it where it says CC-BY on the map instead of linking to the license).

**Comment by keyserj** [comment]
> I think it's fine – if that text clearly says that the map admin(s) can change the license then it's not an issue

> wouldn't be a problem to just specify this in the License page

I agree it seems sufficient if Ameliorate has text explicitly stating that a Topic owner can change the license.

> For better clarity I'd ask on a relevant subreddit or at https://opensource.stackexchange.com/ where you could also mention the site

> If you don't ask about it maybe I will

I think the implications of "owning" don't seem like too big of a deal and that I'll pass on clarifying that for now, but if you end up doing so let me know what you find.

> so that the people need to be credited which if for example some study or news article quotes a node, they should name its author username

> They could control how they're being credited

I hadn't considered these, but they make sense.

> (maybe also some stats like these at some point)

Wow that's a lot of stats! Yeah I hope to add stats at some point - I think they can be cool to see, add more color to history, and can also be motivating for contributors.

> (maybe link to it where it says CC-BY on the map instead of linking to the license)

Yeah that's what I was thinking. Then the docs would link to the thorough license description elsewhere.

**Comment by keyserj** [comment]
Note: the react-flow attribution is currently being slightly covered-up on mobile by some overlayed buttons; we'll need to make sure the CC-BY attribution isn't obstructed and is clickable. Maybe the toolbar & details button shouldn't be overlayed at all on mobile.

![Image](https://github.com/user-attachments/assets/d34498a1-6b22-4455-8d22-cfae81211168)
